spec/unit/typesafe/config/config_value_spec.rb

```diff
@@ -0,0 +1,503 @@
+    expect { m.delete(:a) }.to raise_error(NoMethodError)
+  end
+end
+
+describe "ConfigList" do
+  specify "should implement read only list" do
+    values = ["a", "b", "c"].map { |i| TestUtils.string_value(i) }
+    l = SimpleConfigList.new(TestUtils.fake_origin, values)
+
+    expect(values[0]).to eq(l[0])
+    expect(values[1]).to eq(l[1])
+    expect(values[2]).to eq(l[2])
+
+    expect(l.include? TestUtils.string_value("a")).to be_truthy
+
+    expect(l.index(values[1])).to eq(1)
+
+    expect(l.empty?).to be_falsey
+
+    expect(l.map { |v| v }).to eq(values.map { |v| v })
+
+    expect(l.rindex(values[1])).to eq(1)
+
+    expect(l.size).to eq(3)
+
+    expect { l.push(TestUtils.int_value(3)) }.to raise_error(NoMethodError)
+    expect { l << TestUtils.int_value(3) }.to raise_error(NoMethodError)
+    expect { l.delete(TestUtils.int_value(2)) }.to raise_error(NoMethodError)
+    expect { l.delete(1) }.to raise_error(NoMethodError)
+    expect { l.push(TestUtils.int_value(3)) }.to raise_error(NoMethodError)
+    expect { l[0] = TestUtils.int_value(42) }.to raise_error(NoMethodError)
+  end
+end
+
+describe "Objects throwing ConfigNotResolvedError" do
+  context "ConfigSubstitution" do
+    specify "should throw ConfigNotResolvedError" do
+      expect{ TestUtils.subst("foo").value_type }.to raise_error(ConfigNotResolvedError)
+      expect{ TestUtils.subst("foo").unwrapped }.to raise_error(ConfigNotResolvedError)
+    end
+  end
+
+  context "ConfigDelayedMerge" do
+    let(:dm) { ConfigDelayedMerge.new(TestUtils.fake_origin, [TestUtils.subst("a"), TestUtils.subst("b")]) }
+
+    specify "should throw ConfigNotResolvedError" do
+      expect{ dm.value_type }.to raise_error(ConfigNotResolvedError)
+      expect{ dm.unwrapped }.to raise_error(ConfigNotResolvedError)
+    end
+  end
+
+  context "ConfigDelayedMergeObject" do
+    empty_object = SimpleConfigObject.empty
+    objects = [empty_object, TestUtils.subst("a"), TestUtils.subst("b")]
+
+    let(:dmo) { ConfigDelayedMergeObject.new(TestUtils.fake_origin, objects) }
+
+    specify "should have value type of OBJECT" do
+      expect(dmo.value_type).to eq(Hocon::ConfigValueType::OBJECT)
+    end
+
+    specify "should throw ConfigNotResolvedError" do
+      expect{ dmo.unwrapped }.to raise_error(ConfigNotResolvedError)
+      expect{ dmo["foo"] }.to raise_error(ConfigNotResolvedError)
+      expect{ dmo.has_key?(nil) }.to raise_error(ConfigNotResolvedError)
+      expect{ dmo.has_value?(nil) }.to raise_error(ConfigNotResolvedError)
+      expect{ dmo.each }.to raise_error(ConfigNotResolvedError)
+      expect{ dmo.empty? }.to raise_error(ConfigNotResolvedError)
+      expect{ dmo.keys }.to raise_error(ConfigNotResolvedError)
+      expect{ dmo.size }.to raise_error(ConfigNotResolvedError)
+      expect{ dmo.values }.to raise_error(ConfigNotResolvedError)
+      expect{ dmo.to_config.get_int("foo") }.to raise_error(ConfigNotResolvedError)
+    end
+  end
+end
```

Comment:
I think you're fine, it's just a sanity check to know they're throwing the right error, and doesn't have a lot of functionality behind it